### PR TITLE
Page Scroll Implementation

### PR DESCRIPTION
**Pull request by BrittanyIRL**
# Page Scroll Implementation

[Notion Page](https://www.notion.so/formidable/Ensure-scroll-functionality-1b52193a57ea46db83e97a2d74ec28dd)

## Requirements  ##
1. When clicking a linked heading in the docs page, it should auto-scroll that heading to the top of the page but NOT hidden behind the header
2. Refreshing the page should auto-scroll to the appropriate spot if there is a hash in the url
3. The sidebar nav should behave similarly (see existing site) 

## Description ## 
- update lodash package 
- add react-scroll for animated scrolling: https://github.com/fisshy/react-scroll 
- z-index update for header so that sidebar is positioned where it should be.
- Split out some constants used for scrolling. Ie) calculating true offset/position, default classes to look for, possible routes 
- `ScrollToCurrentSection` replaces the old ScrollToTop component as it is more semantic. Using some hooks to decide when to look for the proper spot to scroll to. `useEffect` watches for the pathname update - pathname only updates on page initial load, it has nothing to do with the hash changes in the page. Then useLayoutEffect is watching for hash updates as well as page container height updates (initially) since the playground elements aren't sized until after first paint which makes the initial scrollTo unreliable. 

### On Refresh or specific hyperlink w/ hash ###
refreshscroll

### In docs scroll ###
inpage scroll

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -1,56 +1,73 @@
...
-      scrollContent(this.props.location);
-      scrollSidebar(this.props.location);
-    }
-  }
+const ScrollToCurrentSection = ({ location, children }) => {
+  const { pathname, hash = "" } = location;
+
+  const [pageContentHeight, setPageContentHeight] = useState();
+
+  const pageContentHeightObserver = new ResizeObserver(element => {
```
Comment (carlos-kelly) [inline src/app.js]: Do we need to polyfill this for Edge and Safari? 
https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver

**Comment by boygirl** [comment]
The scroll that happens when the sidebar is clicked is lovely, but I'm finding the scroll on load / refresh to be jarring.

File: src/app.js (hunk 1 above)
Comment (BrittanyIRL) [reply src/app.js]: yes you are absolutely right, i realized i'd forgotten this as soon as I got on the lightrail last night, adding one this AM. thank you for catching this!!!

[the remark this answers is not in the document]
Comment (BrittanyIRL) [reply src/app.js]: oh and so it does! thank you for double checking that. I will resolve and tag you for rereview.

**Comment by BrittanyIRL** [comment]
> The scroll that happens when the sidebar is clicked is lovely, but I'm finding the scroll on load / refresh to be jarring.

What about this? 
I shortened the animation time. I think it works well? 

faster-scroll-on-load

inpage scroll

**Pull request merged**